DetectorDescription/Core/interface/DDName.h

```diff
@@ -51,44 +42,16 @@ class DDName
    */
   const std::string fullname() const { return ns() + ":" + name(); }
 
-  // DEPRECATED!!!
-  operator std::string() const { return ns() + ":" + name(); }
-
-  id_type id() const { return id_;}
-    
   bool operator<(const DDName & rhs) const { return id_ < rhs.id_; }
   bool operator==(const DDName & rhs) const { return id_ == rhs.id_; }
 
 private:
+  
   id_type id_;
     
-  static Registry::iterator registerName(const std::pair<std::string,std::string> & s);  
-};
-
-//! DDNameInterface provides a common interface to DDD entities
-/** DDLogicalPart, DDMaterial, DDSolids, ... all are uniquely identified by
-    their class and the value of their associated DDName. 
-    DDNameInterface provides read-access to their DDName.
-*/
-struct DDNameInterface
-{
-  virtual ~DDNameInterface() {}
-  
-  //! Returns the \a name without the \a namespace
-  virtual const std::string & name() const=0;   // name without namespace
-  
-  //! Return the \a namespace 
-  virtual const std::string & ns() const=0; // only the namespace 
-  
-  //! Return the DDName
-  DDName ddname() const { return DDName(name(),ns()); } 
-  
-  virtual operator bool() const=0;
-  
-  //! \b don't \b use \b! 
-  virtual int id() const=0; // dont't know if usefull ...
+  static Registry::iterator registerName( const std::pair<std::string, std::string> & s );
```

Comment:
Instead of returning an iterator, which implies one could move forward, why not just return the `std::pair<>`?